src/js/tabs/onboard_logging.js

```
@@ -489,49 +489,59 @@ onboard_logging.initialize = function (callback) {
+
+async function requestBlock(addr) {
+    try {
+        // Await the MSP async read; ensure St.dataflashRead returns a Promise
+        return await St.dataflashRead(addr);
+    } catch (e) {
+        console.error(`Request failed for block at ${addr}:`, e);
+        return null;
+    }
+}
+
+function checkCRC(block) {
```

ESLint

[error] Expected indentation of 20 spaces but found 0.

(indent)